src/Ship.js

```diff
@@ -0,0 +1,34 @@
+function Ship(itinerary){
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+};
+
+Ship.prototype.setSail = function() {
+   const itinerary = this.itinerary;
+   const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+   if (currentPortIndex === (itinerary.ports.length - 1)) {
+    throw new Error('End of itinerary reached')
+   }
+
+   this.previousPort = this.currentPort;
+   this.currentPort.removeShip(this);
+   this.currentPort = null;
+
+   
+};
+
+Ship.prototype.dock = function() {
+  const itinerary = this.itinerary;
+  const previousPortIndex = itinerary.ports.indexOf(this.previousPort);
+
+  this.currentPort = itinerary.ports[previousPortIndex + 1];
+
+  this.currentPort.addShip(this);
```

Comment:
This should 4 levels of indentation to math everything else in the project, you could also condense it since it is a very short function:

```
    const itinerary = this.itinerary;
    const previousPortIndex = itinerary.ports.indexOf(this.previousPort);
    this.currentPort = itinerary.ports[previousPortIndex + 1];
    this.currentPort.addShip(this);
```